Calorimetry/python/jetResolution/histPool.py

```diff
@@ -0,0 +1,151 @@
+  ]
+
+  def __init__(self):
+    """Initialize."""
+    self.histDict2D = {}
+
+  def initNewSetOfHists(self, setName, histInformation):
+    """Initialize new set of histograms."""
+    if setName in self.histDict2D:
+      return True
+    tmpDict = {}
+    for iHistInfo in histInformation:
+      if iHistInfo[0] == 'TH1F':
+        if len(iHistInfo) == 5:
+          tmpArr = array.array('d', iHistInfo[4])
+          tmpHist = ROOT.TH1F(randomString(10), iHistInfo[2], iHistInfo[3],
+                              tmpArr)
+        if len(iHistInfo) == 6:
+          tmpHist = ROOT.TH1F(randomString(10), iHistInfo[2], iHistInfo[3],
+                              iHistInfo[4], iHistInfo[5])
+        tmpDict[iHistInfo[1]] = tmpHist
+    self.histDict2D[setName] = tmpDict
+
+  def createJERFitHistInfo(self):
+    """???"""
+    outHistInfo = []
+    histInfoTemplate = ['TH1F', 'cosThetaBin', '; reco + true neutrino energy [GeV]; Entries',
+                        10000, 0., 500.]
+    for i in range(1, len(self.cosThetaRange)):
+      thetaRange = 'cosTheta: ' + str(self.cosThetaRange[i - 1]) + ' - ' + str(self.cosThetaRange[i])
+      binNumber = i - 1
+      histInfoEntry = list(histInfoTemplate)
+      histInfoEntry[1] += str(binNumber)
+      histInfoEntry[2] = thetaRange + histInfoEntry[2]
+      outHistInfo.append(histInfoEntry)
+    return outHistInfo
+
+  def fillHist(self, histName, val, histSetName=''):
+    """Fill histogram.
+
+    Return True if val is successfully added to histogrm, otherwise return
+    False.
+    """
+    if histSetName not in self.histDict2D:
+      self.initNewSetOfHists(histSetName, self.histInfo)
+      jerFitHistInfo = self.createJERFitHistInfo()
+      fitHistSetName = histSetName + '/fit' if histSetName else 'fit'
+      self.initNewSetOfHists(fitHistSetName, jerFitHistInfo)
+
+    if histName in self.histDict2D[histSetName]:
+      self.histDict2D[histSetName][histName].Fill(val)
+      return True
+    else:
+      print("Can't find the hitogram: %s" % histName)
+      return False
+
+  def fillAllHists(self, vTuple, histSetName=''):
+    """Fill all histograms."""
+    cosTheta = abs(vTuple.quark1.Pz() / vTuple.quark1.P())
+    self.fillHist('cosTheta', cosTheta, histSetName)
+
+    totalTrueE = vTuple.totalTrueVisible + vTuple.totalTrueInvisible
+    self.fillHist('true_totalE', totalTrueE.E(), histSetName)
+
+    genJetSum = vTuple.genJets[0] + vTuple.genJets[1]
+    self.fillHist('gen_InvMass', genJetSum.M(), histSetName)
+    self.fillHist('gen_totalE', genJetSum.E(), histSetName)
+
+    recoJetSum = vTuple.recoJets[0] + vTuple.recoJets[1]
+    self.fillHist('reco_InvMass', recoJetSum.M(), histSetName)
+    self.fillHist('reco_totalE', recoJetSum.E(), histSetName)
+
+    for i in range(1, len(self.cosThetaRange)):
+      if cosTheta > self.cosThetaRange[i - 1] and cosTheta < self.cosThetaRange[i]:
+        binNumber = i - 1
+        fillVal = vTuple.recoJets[0].E() + vTuple.totalTrueInvisible.E()
```

Comment:
typically the amount of invisible energy is negligible, I think maybe calculating the response value between the matched jets might be rather the value to fill, reconstructed-jet E/ matched MC particle-jet E, and for jet 0 and jet 1, then the histogram limits would also need no energy dependent range for CLIC and FCC, we could just have a range from 0 to 2 (or if we believe things to go south up to 3).
Final results are close to each other in both cases, if we want to correct the bias of the invisible energy, maybe a split of it to both jets would be better on average.